### PR TITLE
Updates to pipeline context object

### PR DESCRIPTION
**Pull request by annatisch**

### CODE REVIEW

**Comment by adxsdk6** [comment]
Can one of the admins verify this patch?

**Comment by adxsdk6** [comment]
Can one of the admins verify this patch?

File: azure-core/azure/core/pipeline/__init__.py [hunk 1]
```diff
@@ -45,6 +45,35 @@ def __exit__(self, exc_type, exc_value, traceback):
...
+class PipelineContext(dict):
+
+    def __init__(self, transport, **kwargs):
+        self.transport = transport
+        self.options = kwargs
+        self._protected = ['transport', 'session', 'options']
+
+    def __setitem__(self, key, item):
+        if key in self._protected:
+            raise ValueError('Context value {} cannot be overwritten.'.format(key))
```
Comment (bryevdv) [inline azure-core/azure/core/pipeline/__init__.py]: `session` is a protected attr but its not set in the initializer and can't be set with `__setitem__` either, what sets it?

File: azure-core/azure/core/pipeline/transport/aiohttp.py [hunk 2]
```diff
@@ -47,60 +47,64 @@
...
-    def _init_session(self, session):
-        pass  # configure sesison
-
-    def build_context(self, **kwargs):
-        # type: () -> AioHttpContext
-        return AioHttpContext(session=self.session, transport=self, **kwargs)
+        self.session = None
+        self.config.connection.keep_alive = False
+
+    def _build_ssl_confg(self, **config):
```
Comment (bryevdv) [inline azure-core/azure/core/pipeline/transport/aiohttp.py]: typo, `_build_ssl_config` ?

File: azure-core/azure/core/pipeline/__init__.py (hunk 1 above)
Comment (annatisch) [reply azure-core/azure/core/pipeline/__init__.py]: Oops - forgot to remove it from that list - thanks!

File: azure-core/azure/core/pipeline/transport/aiohttp.py (hunk 2 above)
Comment (annatisch) [reply azure-core/azure/core/pipeline/transport/aiohttp.py]: Oh good catch

**Pull request merged**